app/dao/services_dao.py

```diff
@@ -372,8 +386,18 @@ def _delete(query):
     _delete(NotificationHistory.query.filter_by(service=service))
     _delete(Notification.query.filter_by(service=service))
     _delete(Job.query.filter_by(service=service))
-    _delete(Template.query.filter_by(service=service))
-    _delete(TemplateHistory.query.filter_by(service_id=service.id))
+
+    templates = Template.query.filter_by(service=service)
+    for template in templates:
+        template.folder = None
+        _delete(TemplateHistory.query.filter_by(id=template.id))
```

Comment:
would it be easier to just delete these template history objects in one row after the for loop? means we only issue one query rather than lots, and also just in case we end up with templates being deleted but the history still being around, then it prevents any weird issues.

so

```
    templates = Template.query.filter_by(service=service)
    for template in templates:
        template.folder = None
    _delete(templates)
    _delete(TemplateHistory.query.filter_by(service_id=service.id))
```